### DISCUSSION TITLE
Water flow calibration drift

### DISCUSSION THREAD

**Discussion by jac2424** [tronikos/esphome-magnetometer-water-gas-meter#18] [General]
Just wondering if anyone else has issue with water usage calculations staying calibrated and consistent as mine seems to drift day to day.  Just when I have everything set with proper  usage logged for the day, next day it becomes in accurate again. Is anyone else noticing similar measurement/calibration drift?

**Comment by tronikos** [comment]
Does the range of the magnetic field change? Is your sensor securely mounted?

**Reply by jac2424** [reply]
Sensor is securely mounted with double sided tape,  may look into mounting with something else down the road. The magnetic field levels don’t seems to change. Perhaps my issue it’s close to a dryer and maybe that’s impacting the reading somehow when it  runs. Also I have the sensor being powered by 3.3v from ESP32, will switch to 5v to see if it helps.

**Reply by nosoftwareskills** [reply]
My sensor is mounted with double sided tape as I test this out. Once I get good readings I will find a better way to secure it.

What I am seeing is the magnetic field for x,y, and z changes through the day. The difference between max and min is mostly consistent, but the average between them goes up and down. If I hit calibrate everything is good for about 10 minutes then I need to hit calibrate again. 

I've tried a couple different sensors (from the same order) and it's the same thing. I haven't seen any similar comments. Anyone have thoughts what this could be? Does the sensor need calibration? Maybe these are bad sensors? TIA
![Screenshot_20241023-213822](https://github.com/user-attachments/assets/782a532f-e625-4d06-a456-f1453b272339)

**Reply by jac2424** [reply]
> My sensor is mounted with double sided tape as I test this out. Once I get good readings I will find a better way to secure it.
> 
> What I am seeing is the magnetic field for x,y, and z changes through the day. The difference between max and min is mostly consistent, but the average between them goes up and down. If I hit calibrate everything is good for about 10 minutes then I need to hit calibrate again.
> 
> I've tried a couple different sensors (from the same order) and it's the same thing. I haven't seen any similar comments. Anyone have thoughts what this could be? Does the sensor need calibration? Maybe these are bad sensors? TIA ![Screenshot_20241023-213822](https://private-user-images.githubusercontent.com/158687014/379513279-782a532f-e625-4d06-a456-f1453b272339.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3MzgyOTksIm5iZiI6MTcyOTczNzk5OSwicGF0aCI6Ii8xNTg2ODcwMTQvMzc5NTEzMjc5LTc4MmE1MzJmLWU2MjUtNGQwNi1hNDU2LWYxNDUzYjI3MjMzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNFQwMjQ2MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lOTUwODg1ZThmMmY0NDA4OWExYzU4ZGY3ZDMwZGJmYjFkZTY2YzhlMzM1OWQ5NGFiMDY0MzBlZmM1OWQxOTc0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UbMxbzaLPkRp3nVY0M2YPS1yokeNi8gGio3Ni53ww-E)

Mine is still mounted with double sided tape but just moved the orientation of the senor slightly so its not facing the dryer side which was also causing temp spikes on the temp sensor on the ESP32. I did also replace the wiring with CAT6 shielded wire instead of the thermostat and it seems to be pretty stable and reading the water fairly accurately (within +/- 0.25 m3 of the actual meters totals).

**Reply by nosoftwareskills** [reply]
I'm glad to hear it is working ok now. It sounds like your issue was fluctuations in temperature. Maybe you want to close this? I opened a separate conversation for what I am seeing.

**Comment by jac2424** [comment]
Resolved and working now.
